packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -34,24 +38,30 @@ export class DataProxyEngine extends Engine {
     this.env = this.config.env ?? {}
     this.inlineSchema = config.inlineSchema ?? ''
     this.inlineDatasources = config.inlineDatasources ?? {}
+    this.inlineSchemaHash = config.inlineSchemaHash ?? ''
+    this.clientVersion = config.clientVersion ?? 'unknown'
 
     this.logEmitter = new EventEmitter()
     this.logEmitter.on('error', () => {})
 
-    this.initPromise = this.init()
-  }
-
-  /**
-   * !\ Asynchronous constructor that inits the properties marked with `!`.
-   * So any function that uses such a property needs to await `initPromise`.
-   */
-  private async init() {
-    // we set the network stuff up for the engine to make http calls to the proxy
     const [host, apiKey] = this.extractHostAndApiKey()
-    this.schemaHash = await createSchemaHash(this.inlineSchema)
-    this.clientVersion = getClientVersion(this.config)
+    this.remoteClientVersion = getClientVersion(this.config)
     this.headers = { Authorization: `Bearer ${apiKey}` }
     this.host = host
+
+    const promise = Promise.resolve() // hack for cloudflare
```

Comment:
What is the hack here?

Reply:
Clouflare errored with "promise will never terminate" or something similar. I'll remove it since my repro does not fail anymore.

Reply:
Correction: it still happens:
```
[Error] Some functionality, such as asynchronous I/O, timeouts, and generating random values, can only be performed while handling a request.
```

That's because we instantiate the client outside of the request handler. This essentially prevents immediate execution of the promise.

Reply:
Can we put this explanation to the code?